### DISCUSSION TITLE
Display original size centered image with centered-caption

### DISCUSSION THREAD

**Discussion by kulbhushanchand** [mmistakes/minimal-mistakes#3018] [Q&A] [answered]
As per the [documentation](https://raw.githubusercontent.com/mmistakes/minimal-mistakes/master/docs/_posts/2013-01-10-markup-image-alignment.md)

The small size figure is unnecessarily stretched as shown in example below. 
**How can I display original size centered image with centered-caption?**

```
![website-lighthouse-metrics]({{ site.url }}{{ site.baseurl }}/assets\images\posts\website-lighthouse-metrics-desktop.jpg){: .align-center}

<figure class="align-center">
  <img src="{{ site.url }}{{ site.baseurl }}/assets\images\posts\website-lighthouse-metrics-desktop.jpg" alt="">
    <figcaption>This image is unnecessarily stretched</figcaption>
</figure> 
```

![image](https://user-images.githubusercontent.com/17742733/123463278-f4330980-d608-11eb-966f-0a519d02098c.png)

**Comment by mmistakes** [comment, marked as answer]
Start by inspecting the element you want to change in your browser's web dev tools.
It will show all the CSS being applied which will be a clue to what needs to be changed.

TL/DR there's `width: 100%` being applied to `figure img`. Remove that and or add `max-width: 100%` and it won't scale them up... only down.

https://github.com/mmistakes/minimal-mistakes/blob/master/_sass/minimal-mistakes/_base.scss#L224-L229

**Reply by kulbhushanchand** [reply]
Thanks, but now the image is now left-aligned.
![image](https://user-images.githubusercontent.com/17742733/123465689-124e3900-d60c-11eb-9a99-718abd5bd97e.png)

**Reply by mmistakes** [reply]
https://css-tricks.com/centering-css-complete-guide/

**Reply by kulbhushanchand** [reply]
Thanks I'll check this out.

**Reply by kulbhushanchand** [reply]
In case anyone else wants to achieve the same, here is what I have changed

As per the suggestion above at https://github.com/mmistakes/minimal-mistakes/discussions/3018#discussioncomment-922267
```
  img {
    max-width: 100%; // original is width: 100%;
    display: block;
    margin-left: auto;
    margin-right: auto;
    border-radius: $border-radius;
    -webkit-transition: $global-transition;
    transition: $global-transition;
  }
```

where, `max-width: 100%;` is to preserve the size and 
```
    display: block;
    margin-left: auto;
    margin-right: auto;
```
is to center the image. 

Thanks to @mmistakes for the kind help

**Reply by kulbhushanchand** [reply]
@mmistakes Although I got this working as you suggested, but in the theme the `class="align-center"` does not seems to work with `<figure>` element as I have experienced in the original question.
I would be thankful if you can test it at your convenience.

**Comment by mmistakes** [comment]
The align classes only work on inline elements like `img`

`figure` is a block element which is why you need to use the margin `auto` to center it, that doesn’t work on `inline`.

**Reply by mmistakes** [reply]
Scratch that. The align class works on a `figure`, just doesn’t cascade down to the image inside of it. You need to center that as you have above.
